controllers/osbuild_controller.go

```diff
@@ -47,16 +81,220 @@ type OSBuildReconciler struct {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("osbuild", req.Name)
+
+	osBuild, err := r.OSBuildRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, err
+	}
+
+	if osBuild.DeletionTimestamp != nil {
+		// The OSBuild CRs that were created by that OSBuildConfig would be deleted
+		// thanks to setting controller reference for each OSBuild CR
+		return ctrl.Result{}, nil
+	}
+
+	var composeStatus composer.ComposeStatusValue
+	containerBuildStatus := osbuildv1alpha1.OSBuildConditionType("")
+	if osBuild.Status.Conditions != nil {
+		conditionLen := len(osBuild.Status.Conditions)
+		containerBuildStatus = osBuild.Status.Conditions[conditionLen-1].Type
+	}
+
+	if osBuild.Status.ContainerComposeId != EmptyComposeID && containerBuildStatus != ContainerBuildDone {
+		logger.Info("update the edge-container's compose ID job status")
+		composeStatus, err := r.UpdateContainerComposeId(ctx, logger, osBuild)
+		if err != nil {
+			logger.Error(err, "failed to get compose ID status")
+			return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, nil
+		}
+
+		if composeStatus == composer.ComposeStatusValuePending {
+			logger.Info(fmt.Sprintf("the job ID %s, is still in progress", osBuild.Status.ContainerComposeId))
+			return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, nil
+		}
+	} else {
+		// build edge container
+		logger.Info("create an edge-container")
+		err = r.postComposeEdgeContainer(ctx, logger, osBuild)
+		if err != nil {
+			logger.Info("failed to create an edge-container")
+			return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, nil
+		}
+		logger.Info("new job created for edge-container, requeue to sample its status")
+		return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, nil
+	}
 
-	// TODO(user): your logic here
+	if osBuild.Status.IsoComposeId != EmptyComposeID {
+		logger.Info("update the edge-installer's compose ID job status")
+		composeStatus, err := r.UpdateIsoComposeId(ctx, logger, osBuild)
+		if err != nil {
+			logger.Error(err, "failed to get compose ID status")
+			return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, nil
+		}
+
+		if composeStatus == composer.ComposeStatusValuePending {
+			logger.Info(fmt.Sprintf("the job ID %s, is still in progress", osBuild.Status.IsoComposeId))
+			return ctrl.Result{Requeue: true, RequeueAfter: RequeueForTwoMinutes}, nil
+		}
+	} else if composeStatus == composer.ComposeStatusValueSuccess && osBuild.Spec.Details.TargetImage.TargetImageType == EdgeInstallerImgType {
+		// TODO postComposeEdgeInstaller
+	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *OSBuildReconciler) UpdateContainerComposeId(ctx context.Context, logger logr.Logger, osBuild *osbuildv1alpha1.OSBuild) (composer.ComposeStatusValue, error) {
```

Comment:
I think updateContainerComposeStatus is a better name since the method updates the status, not the id